### PR TITLE
refactor ContentPackInstaller a little, tests, clearer error

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-39975)

## Description
main thing here i added is the better error messaging and handling, i also so that we had added
if type(res) is list:
    res = res[0]
only in a couple of cases which is what partially caused this customer's issue so i made 1 function to call execute_command

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/ContentPackInstaller/ContentPackInstaller.py [hunk 1]
```diff
@@ -358,6 +340,7 @@ def main():
         )
 
     except Exception as e:
+        demisto.debug(f'{e}')
```
Comment (JudahSchwartz) [inline Packs/CommonScripts/Scripts/ContentPackInstaller/ContentPackInstaller.py:343]:
```suggestion
        demisto.debug(f'error occured during script execution {e}')
```

File: Packs/CommonScripts/Scripts/ContentPackInstaller/ContentPackInstaller.py [hunk 2]
```diff
@@ -22,30 +22,33 @@ def __init__(self, instance_name: str = None):
...
         self.get_installed_packs()
 
-    def get_installed_packs(self) -> None:
-        """Gets the current installed packs on the machine.
-        """
-        demisto.debug(f'{SCRIPT_NAME} - Fetching installed packs from marketplace.')
-
-        args = {'uri': '/contentpacks/metadata/installed'}
-
+    def _call_execute_command(self, args, command):
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/ContentPackInstaller/ContentPackInstaller.py:25]: Let's be more aligned with the execute command function
```suggestion
    def _call_execute_command(self, command, args):
```

**Review by ShahafBenYakir — APPROVED** [review]
Great.
I know it's a bit tedious but let's add docstring to the tests. If the test is very straightforward and there are no cases then there's no need to add "given, when, then".

**Comment by JudahSchwartz** [comment]
adding force due to 
`[BC108] - You can't delete current marketplaces or add new ones if doing so will remove existing ones. Please undo the change or request a forced merge.`

xpanse isnt supported by core rest api. so this script shouldnt be in xpanse either

**Pull request merged**